eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1244,7 +1244,14 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
 
     case Event(WatchFundingSpentTriggered(tx), d: DATA_SHUTDOWN) => handleRemoteSpentOther(tx, d)
 
-    case Event(c: CMD_CLOSE, d: DATA_SHUTDOWN) => handleCommandError(ClosingAlreadyInProgress(d.channelId), c)
+    case Event(c: CMD_CLOSE, d: DATA_SHUTDOWN) =>
+      c.feerates match {
+        case Some(feerates) if c.feerates != d.closingFeerates =>
+          log.info("updating our closing feerates: {}", feerates)
+          handleCommandSuccess(c, d.copy(closingFeerates = c.feerates)) storing()
+        case _ =>
+          handleCommandError(ClosingAlreadyInProgress(d.channelId), c)
+      }
```

Comment:
You're not allowed to change the script, as this would lead to undefined behavior for your peer, so I think it's simpler to keep it like this. I'll also add a check on the script, if it's different from the previous one I'll return a specific error.

Reply:
At least let's make sure that no script is provided, so it's less confusing.

```suggestion
      c.feerates match {
        case Some(feerates) if c.feerates != d.closingFeerates && c.scriptPubKey.isEmpty =>
          log.info("updating our closing feerates: {}", feerates)
          handleCommandSuccess(c, d.copy(closingFeerates = c.feerates)) storing()
        case _ =>
          handleCommandError(ClosingAlreadyInProgress(d.channelId), c)
      }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1768/commits/78bb837e3ee767bfac534ea5c057627d0fbe79eb